File: seungyeop/22_12_28_02.java
import java.util.*;

class Solution {
    public int solution(int[] ingredient) {
        Stack<Integer> stack = new Stack<>();  //stack 만들기
        int count = 0;
        for (int i : ingredient) {
            stack.add(i);
            if (stack.size() >= 4 && stack.elementAt(stack.size() - 1) == 1
                && stack.elementAt(stack.size() - 2) == 3
                && stack.elementAt(stack.size() - 3) == 2
                && stack.elementAt(stack.size() - 4) == 1) {
                count++;
                stack.pop();stack.pop();stack.pop();stack.pop();
            }
        }
        return count;
    }
}

File: yejin/22_12_28_02.py
# 햄버거 만들기

# 시도 1
# 66.7 / 100.0
def solution(ingredient):
    count = 0
    a= ''.join(str(s) for s in ingredient)       
    while "1231" in a: 
        count+= a.count("1231")
        a= a.replace("1231","")
    return count

# join을 이용해 문자열로 다 이어붙이고
# (1231이 없어질때까지) while 이용?    
    # '1231'의 개수를 count하고 
    # replace로 1231을 없앤다.
# count를 return한다.

# 시도2
def solution(ingredient):
    count = 0
    a= ''.join(str(s) for s in ingredient)        
    while "1231" in a: 
        a= a.replace("1231"," ")
        count+= a.count(" ")
        a= a.replace(" ","")
    return count
  
#   실패다. 다른 방법을 고민해보자.
